### PR TITLE
Convert /wallet/encrypt to /wallet/init

### PR DESCRIPTION
**Pull request by bitspill**
Untested but I see no reason this shouldn't work... let me know if I did something silly.

### CODE REVIEW

**Comment by bitspill** [comment]
Forgot to reference the issue: #754

File: api/wallet.go [hunk 1]
```diff
@@ -196,18 +196,28 @@ func (srv *Server) walletEncryptHandlerPOST(w http.ResponseWriter, req *http.Req
...
+func (srv *Server) walletInitHandler(w http.ResponseWriter, req *http.Request) {
+	if req.Method == "POST" {
+		srv.walletInitHandlerPOST(w, req)
+		return
+	}
+	writeError(w, "unrecognized method when calling /wallet/init", http.StatusBadRequest)
+}
+
 // walletEncryptHandler handles API calls to /wallet/encrypt.
+// Replaced with /wallet/init kept for compatability
```
Comment (DavidVorick) [inline api/wallet.go]: first word of the docstring should be the function name.

Also, need to add a COMPAT message.

Something like this:

```
// walletEncryptHandler is a legacy alias for walletInitHandler.
//
// COMPAT v0.4.0

File: api/api.go [hunk 2]
```diff
@@ -102,6 +102,7 @@ func (srv *Server) initAPI(addr string) {
 		handleHTTPRequest(mux, "/wallet", srv.walletHandler)                           // GET
 		handleHTTPRequest(mux, "/wallet/address", srv.walletAddressHandler)            // GET
 		handleHTTPRequest(mux, "/wallet/backup", srv.walletBackupHandler)              // POST
+		handleHTTPRequest(mux, "/wallet/init", srv.walletInitHandler)                  // POST
 		handleHTTPRequest(mux, "/wallet/encrypt", srv.walletEncryptHandler)            // POST
```
Comment (DavidVorick) [inline api/api.go]: need to add a COMPAT tag

`handleHTTPRequest(mux, "/wallet/init", srv.walletInitHandler)                  // POST - COMPATv0.4.0`

**Comment by bitspill** [comment]
Aye! It passes.

**Comment by bitspill** [comment]
Just squashed those messy commits.

**Pull request merged**